core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -217,6 +217,21 @@ final case class EmptyStringFound(typ: String, location: Option[ConfigValueLocat
 }
 
 /**
+ * A failure representing an unexpected empty traversable
+ *
+ * @param typ the type that was attempted to be converted to from an empty string
+ * @param location an optional location of the ConfigValue that raised the
+ *                 failure
+ * @param path an optional path to the value which was an unexpected empty string
+ */
+final case class EmptyTraversableFound(typ: String, location: Option[ConfigValueLocation], path: Option[String]) extends ConfigReaderFailure {
```

Comment:
Right now, `ConfigReaderFailure` is `sealed`, which forces us to put all implementations in the same file. I don't believe it will be reasonable to continue adding new failures as the number of modules grow, each one with their (rightfully) specific errors. Moreover, users may want to have their own failures and right now they have to settle with one of the existing ones.

I know this one is not particularly specific to cats, but unless `fromNonEmpty` is moved to the core, I think this should be defined in `pureconfig-cats` too. Nevertheless, we should "unseal" `ConfigReaderFailure` and allow custom implementations anywhere.